xapian-core/tests/api_backend.cc

```diff
@@ -1401,3 +1401,35 @@ DEFINE_TESTCASE(embedded1, singlefile) {
 
     return true;
 }
+
+static string get_db_table_path(const string& db_path){
+    string db_table_path;
+    if(endswith(get_dbtype(), "glass")){
+        db_table_path = db_path + "/postlist.glass";
+    }else if(endswith(get_dbtype(), "chert")){
+        db_table_path = db_path + "/postlist.DB";
+    }
```

Comment:
I think you really just want equality tests for the db type - the multi and remote backends won't work with this test.

And something sensible ought to happen if the dbtype is not one that's expected.  Calling `FAIL_TEST` would do.